cartridges/int_algolia/cartridge/scripts/services/algoliaServiceHelper.js

```diff
@@ -143,6 +143,147 @@ function callJsonService(title, service, params) {
     return callStatus;
 }
 
-module.exports.callService = callService;
-module.exports.callJsonService = callJsonService;
-module.exports.UNEXPECTED_ERROR_CODE = UNEXPECTED_ERROR_CODE;
+/**
+ * @param {string} indexName - full index name (e.g. "testPrefix__products__en_US")
+ * @param {string} pattern   - pattern from the key's indexes field ("*", "myPrefix_*", etc.)
+ * @returns {boolean}        - true if the indexName matches the wildcard or exact pattern
+ */
+function matchesIndexName(indexName, pattern) {
+    // Escape regex special chars except '*'
+    var escaped = pattern.replace(/[.+?^${}()|[\]\\]/g, '\\$&');
+    // Replace '*' with '.*'
+    escaped = escaped.replace(/\*/g, '.*');
+    var re = new RegExp('^' + escaped + '$');
+    return re.test(indexName);
+}
+
+/**
+ * Validate an Algolia API Key's ACLs and index restrictions.
+ * 
+ * @param {dw.svc.Service} service - Service instance to call
+ * @param {string} applicationID   - Algolia Application ID
+ * @param {string} apiKey          - Key we want to validate (admin or search)
+ * @param {string} indexPrefix     - The custom prefix used by the cartridge
+ * @param {Array} locales          - Locales from BM form
+ * @param {boolean} isAdminKey     - If true, we require admin ACLs; if false, we require 'search'
+ * @param {boolean} isRecommendationEnabled - If true, we require 'recommend' ACL
+ * @param {boolean} isContentSearchEnabled - If true, we require 'contentSearch' ACL
+ * @returns {Object} { error:Boolean, errorMessage:String, warning:String }
+ */
+function validateAPIKey(service, applicationID, apiKey, indexPrefix, locales, isAdminKey, isRecommendationEnabled, isContentSearchEnabled) {
+    // 0) Build the required ACL array
+    var requiredACLs;
+    if (isAdminKey) {
+        requiredACLs = ['addObject', 'deleteObject', 'deleteIndex', 'settings'];
+    } else {
+        // minimal read usage is  "search", (initial values for search ke : "search", "listIndexes", "settings)
+        requiredACLs = ['search'];
+
+        if (isRecommendationEnabled) {
+            requiredACLs.push('recommendation');
```

Comment:
Where did you see that was needed? Did you had issue while testing?
AFAIK the recommendations are working find with the Search key.